### DISCUSSION TITLE
Programatically create first admin user?

### DISCUSSION THREAD

**Discussion by benallfree** [pocketbase/pocketbase#1354] [Q&A]
Any way to do this without a custom build?

**Comment by ganigeorgiev** [comment]
At the moment there are 2 options _(cli command(s) for managing admins will be added in the near future)_:

- send `POST /api/admins/create` (`pb.admins.create()`) request (the endpoint is public if there are no admins yet)

- create the admin inside a JS migration
    - create a blank migration file `./pocketbase migrate create "create_admin"`
    - change the created migration file content to something like the following:
      ```js
      migrate((db) => {
        const admin = new Admin()

        admin.email = "test@example.com"
        admin.setPassword("1234567890");

        return Dao(db).saveAdmin(admin)
      }, (db) => {
        const dao = new Dao(db)

        const admin = dao.findAdminByEmail("test@example.com")

        return dao.deleteAdmin(admin)
      })
      ```
      _(I haven't tested the above; all `pb_migrations` migrations are applied automatically on `serve`)_
